packages/sync/src/class-main.php

```diff
@@ -20,6 +20,7 @@ class Main {
 	 * @action plugins_loaded
 	 */
 	public static function configure() {
+		add_action( 'jetpack_user_authorized', array( 'Automattic\\Jetpack\\Sync\\Actions', 'do_initial_sync' ), 10, 0 );
```

Comment:
Within do_initial_sync there is a sync_allowed() check.

I'd recommend moving this below the check (line 25) so that future developers don't have to visit do_initial_sync to understand why it doesn't need to adhere to the disabling of the sync.

Yes I'm nit-picking.

Reply:
> Yes I'm nit-picking.

I'd expect nothing less! I think it's part of the creed.

Reply:
sadly, if we do move it below, the hook will never happen.

it's a chicken and egg thing. when we are within the `plugins_loaded` filter
`sync_allowed` allowed is false ( it's only true when we are in the `jetpack_user_authorized` filter ). So the `jetpack_user_authorized` hook never gets declared.

Reply:
I amended the commit to make it a bit more clear. Let me know your thoughts?